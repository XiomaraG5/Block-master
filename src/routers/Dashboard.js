import {
    Routes, Route,
    Navigate
} from 'react-router-dom';
import { NavBarPrivate } from '../components/NavBarPrivate';
import Home from '../containers/Home';
import MasValoradas from '../containers/MasValoradas';
import MenosValoradas from '../containers/MenosValoradas';
import Mispeliculas from '../containers/Mispeliculas';

export const DashboardRoutes = () => {
    return ( 
        <>
        <NavBarPrivate />
        <Routes>
            <Route path="/home" element={<Home />} />
            <Route path="/mas" element={<MasValoradas />} />
            <Route path="/menos" element={<MenosValoradas />} />
            <Route path="/mis" element={<Mispeliculas />} />
            <Route path='*' element={<Navigate to="/home" />} />
        </Routes>
        </>
    )
}
