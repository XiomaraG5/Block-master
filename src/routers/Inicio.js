import {
    Routes, Route,
    Navigate
} from 'react-router-dom';
import { NavBarPublic } from '../components/NavbarPublic';
import Login from '../containers/Login';
import Registro from '../containers/Registro';

export const Inicio = () => {
    return ( 
        <>
            <div>
                <NavBarPublic />
                <Routes>
                    <Route path="/singin" element={<Login />} />
                    <Route path="/registro" element={<Registro/>} />
                    <Route path='*' element={<Navigate to="/login/singin" />} />
                </Routes>
            </div>

        </>
    )
}
