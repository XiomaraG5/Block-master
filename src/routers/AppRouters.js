import { getAuth, onAuthStateChanged } from 'firebase/auth';
import React, { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { dispatchUser, gettingUserAsync } from '../redux/actions/actionUserMovies';
import { DashboardRoutes } from './Dashboard';
import { Inicio } from './Inicio';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

function AppRouter() {
 const dispatch= useDispatch()
   const [checking, setChecking] = useState(true)
  const [isLoggedIn, setIsLoggedIn] = useState(false)
 
  
  useEffect(() => {
    
    const auth = getAuth();
    onAuthStateChanged(auth, (user) => {

    if(user?.uid){
      gettingUserAsync(user)
      dispatch (dispatchUser(user))
      setIsLoggedIn(true)
    
    }else{
      setIsLoggedIn(false)
      }
      setChecking(false)
      })
  }, [setIsLoggedIn,setChecking])

  if(checking){
    
    return(
        <h1>Espere...</h1>
    )
  }

  return <BrowserRouter>
            <Routes>
      <Route path='/login/*' element={
      <PublicRoute isAuthenticated={isLoggedIn} >
        <Inicio/>
      </PublicRoute> } />

      <Route path='/*' element={
      <PrivateRoute isAuthenticated={isLoggedIn}>
        <DashboardRoutes/>
      </PrivateRoute> } />

      
      </Routes>
     </BrowserRouter>;
}

export default AppRouter;
