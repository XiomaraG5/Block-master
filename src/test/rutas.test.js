import "@testing-library/jest-dom";
import React from "react";
import { Navigate } from "react-router-dom";
import { DashboardRoutes } from "../routers/Dashboard";
import { PrivateRoute } from "../routers/PrivateRoute";
import { PublicRoute } from "../routers/PublicRoute";
import { shallow, mount, render } from 'enzyme';
import AppRouter from "../routers/AppRouters";
import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Inicio } from "../routers/Inicio";

configure({ adapter: new Adapter() });

describe("verificar rutas", () => {
  test("verificar ruta privada", () => {
    const loggin = true
   const wrapper = shallow(<PrivateRoute isAuthenticated={loggin} children={DashboardRoutes} />)
   expect(wrapper).toMatchSnapshot()
   console.log(wrapper);
    
    
  });
  test("verificar ruta publica",()=>{
      const loggin = false
      const wrapper = shallow(<PublicRoute isAuthenticated={loggin} children={Inicio}/>)
      expect(wrapper).toMatchSnapshot()
  })
});

