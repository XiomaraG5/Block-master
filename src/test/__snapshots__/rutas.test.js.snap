// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verificar rutas verificar ruta privada 1`] = `ShallowWrapper {}`;

exports[`verificar rutas verificar ruta publica 1`] = `ShallowWrapper {}`;
